plugin/src/compilation/decompiler.ml

```diff
@@ -140,85 +193,107 @@ let try_rel (trm : constr) : int option =
   | Rel i -> Some i
   | _ -> None
 
-(* Guard fails if the rel index occurs in the term. *)
-let noccur (i : int) (trm : constr) : unit option =
-  if Vars.noccurn i trm then Some () else None
+(* Monadic guard for option. *)
+let guard (b : bool) : unit option =
+  if b then Some () else None
 
-let rec first_pass env trm =
+(* Performs the bulk of decompilation on a proof term. 
+   Returns a list of tactics. *)
+let rec first_pass env sigma trm =
```

Comment:
It's unusual to take in an `evar_map` (`sigma`) and not update it and return it. What is the `evar_map` currently used for, and why does `f` not return an updated `evar_map`?

Reply:
`sigma` is used for `deconstruct_eliminator`, and then updated and passed down when decompiling the rest of the proof.
Since the overall flow is `term ---> tactic list`, I wasn't sure whether or not an updated `evar_map` should be kept at the end?
I also pass an `evar_map` into `tac_to_string` to be able to use `Printer.pr_constr_env`, which just returns a string so I don't think it's ever updated there.